base/array.jl

```diff
@@ -228,6 +228,8 @@ convert{T,n}(::Type{Array{T}}, x::Array{T,n}) = x
 convert{T,n}(::Type{Array{T,n}}, x::Array{T,n}) = x
 convert{T,n,S}(::Type{Array{T}}, x::Array{S,n}) = convert(Array{T,n}, x)
 convert{T,n,S}(::Type{Array{T,n}}, x::Array{S,n}) = copy!(similar(x,T), x)
+convert{T,n,S}(::Type{Array{T}}, X::AbstractArray{S, n}) = copy!(Array(T, size(X)), X)
```

Comment:
Agreed. I submitted an updated PR. 

On another note, even the convert{T,n} methods could be wrapped into these two methods. But, we would loose the type checking for same types. 

One observation, is that the first pair of convert methods do not make a copy, but the new ones do.